### PR TITLE
Exclude PPC64 from the polygonBlock and brushBlock assertions

### LINKED ISSUE #528: Game crashes upon starting any map or campaign
#### Host specs

OS: Fedora 33 (ppc64le)
GCC: 10.2.1 20201125
glibc: 2.32

#### Compiler Options:
```
option(FORCE_COLOR_OUTPUT: ON)
option(COMPILE_COMMANDS: OFF)
option(USE_MFC_TOOLS: OFF)
option(USE_PRECOMPILED_HEADERS: ON)
option(SDL2: ON)
option(OPENAL: ON
option(FFMPEG: ON)
option(BINKDEC: OFF)
option(USE_VULKAN: ON)
option(SPIRV_SHADERC: ON)
option(ONATIVE: ON)
option(USE_SYSTEM_ZLIB: OFF)
option(USE_SYSTEM_LIBPNG: OFF)
option(USE_SYSTEM_LIBGLEW: OFF)
option(USE_SYSTEM_RAPIDJSON: OFF)
```

##### Describe

Game crashes when reaching the loading screen to any campaign or Dev map

```
--------- Game Map Shutdown ----------
--------------------------------------
CleanupShell();
delete loadGUI;
ImGuiHook::Destroy();
delete renderWorld;
delete soundWorld;
delete menuSoundWorld;
session->ShutdownSoundRelatedSystems();
session->Shutdown();
game->Leaderboards_Shutdown();
uiManager->Shutdown();
soundSystem->Shutdown();
AL lib: (WW) alSetError: Error generated on context 0x37feb760, code 0xa001, "Invalid source ID 0"
AL lib: (WW) alSetError: Error generated on context 0x37feb760, code 0xa001, "Invalid source ID 0"
AL lib: (WW) alSetError: Error generated on context 0x37feb760, code 0xa001, "Invalid source ID 0"
AL lib: (WW) ReleaseALBuffers: (0x38892f40) Deleted 551 Buffers
usercmdGen->Shutdown();
eventLoop->Shutdown();
declManager->Shutdown();
renderSystem->Shutdown();
idRenderSystem::Shutdown()
Sys_ShutdownInput: SDL joystick not initialized. Nothing to close.
Shutting down Vulkan subsystem
commonDialog.Shutdown();
UnloadGameDLL();
------------ Game Shutdown -----------
--------- Game Map Shutdown ----------
--------------------------------------
Shutdown event system
--------------------------------------
saveFile.Clear( true );
stringsFile.Clear( true );
CloseLogFile();
fileSystem->Shutdown( false );
Sys_Shutdown();
console->Shutdown();
idKeyInput::Shutdown();
cvarSystem->Shutdown();
cmdSystem->Shutdown();
ClearWarnings( GAME_NAME " shutdown" );
warningCaption.Clear();
errorList.Clear();
idLib::ShutDown();
shutdown terminal support
Trace/breakpoint trap (core dumped)
```

Full stack trace: https://gist.github.com/runlevel5/fe05a1cc7ac0c34563eee89a875f05cc

### PR DESCRIPTION
**Pull request by runlevel5**
Fixes #528

### CODE REVIEW

File: neo/cm/CollisionModel_load.cpp [hunk 1]
```diff
@@ -3653,7 +3653,7 @@ cm_model_t* idCollisionModelManagerLocal::LoadBinaryModelFromFile( idFile* file,
 	assert( model->polygonRefBlocks == NULL || ( model->polygonRefBlocks->next == NULL && model->polygonRefBlocks->nextRef == NULL ) );
 
 	// RB: FIXME
-#if !defined(__x86_64__) && !defined(_WIN64)
+#if !defined(__x86_64__) && !defined(_WIN64) && !defined(__PPC64__)
```
Comment (DanielGibson) [inline neo/cm/CollisionModel_load.cpp:3656]: if this is a general 64bit problem there are probably better ways to address this?
like
```if( sizeof(void*) == 4 )
{
    assert(...);
    assert(...);
}
```
could even use `if constexpr (sizeof(void*) == 4)` if you don't mind requiring C++17

Comment (runlevel5) [reply neo/cm/CollisionModel_load.cpp:3656]: I am not entirely sure, so far I could re-produce that with PPC64BE and PPC64LE. 

Q: Why do we do `!defined(_WIN64)` here? Is it only applicable to non-Windows?

Comment (DanielGibson) [reply neo/cm/CollisionModel_load.cpp:3656]: only applicable to non-64bit-windows, I guess (maybe MSVC doesn't define `__x86_64__`? no idea)

Comment (RobertBeckebans) [reply neo/cm/CollisionModel_load.cpp:3656]: It's true __x86_64__ isn't defined with the VS compiler. I also haven't looked into this for like 8 years. I only recall that those assertions are triggered in 64 bit builds and I'm not sure if it matters without diving deeply in the collision code again.

**Pull request merged**